message/message.go

```diff
@@ -0,0 +1,105 @@
+package message
+
+import "strings"
+
+// Message represents DNS messages
+type Message struct {
+	header  *header
+	queries []*query
+	Answers []*Record
+}
+
+type header struct {
+	ID      uint16
+	QR      byte
```

Comment:
Now that we're looking at this again we realize that we could be consistent about the names we use. byte is an alias for uint8.